### PR TITLE
Verify golangci-lint

### PR DESCRIPTION
**Pull request by marquiz**
Integrate golangci-lint with our build. Fix lint errors.

### CODE REVIEW

File: pkg/cpuid/cpuid_amd64.s [hunk 1]
```diff
@@ -16,9 +16,9 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-TEXT ·cpuidAsm(SB), 4, $0   // 4 = NOSPLIT
-    MOVL    eax_arg+0(FP), AX
-    MOVL    ecx_arg+4(FP), CX
+TEXT ·cpuidAsm(SB), 4, $0-24   // 4 = NOSPLIT
```
Comment (Ethyling) [inline pkg/cpuid/cpuid_amd64.s:19]: Why do we add a `-24` here?

**Comment by Ethyling** [comment]
The CI is not working because `golangci-lint` is misisng. Should we fix this in this PR?

File: pkg/cpuid/cpuid_amd64.s (hunk 1 above)
Comment (marquiz) [reply pkg/cpuid/cpuid_amd64.s]: That was to fix this problem:
```
$ go vet ./pkg/cpuid/
# sigs.k8s.io/node-feature-discovery/pkg/cpuid
pkg/cpuid/cpuid_amd64.s:19:1: [amd64] cpuidAsm: wrong argument size 0; expected $...-24
```

But, good question! There was a more proper way to fix this with `#include "textflag.h"`. Changed accordingly

**Comment by marquiz** [comment]
> The CI is not working because `golangci-lint` is misisng. Should we fix this in this PR?

Yes, stupid mistake. I was too busy filing these to even check that they pass CI 🤦 Should be fixed, now.

**Pull request merged**